packages/expect/src/types.ts

```diff
@@ -152,7 +152,7 @@ export interface Matchers<R extends void | Promise<void>> {
    * Checks that a value is what you expect. It calls `Object.is` to compare values.
    * Don't use `toBe` with floating-point numbers.
    */
-  toBe(expected: unknown): R;
+  toBe(expected: E): R;
```

Comment:
Looking better at this, I'm not sure this is wanted in all the cases. :thinking: 
For example.

1. `toBe`, in theory, should not do type check (e.g. `expect('1').toBe(1)`) because there is `toStrictEqual` for that case.
2. Other matches may want type check as well, for consistency, but that makes things more complex. (e.g. `toHaveReturnedWith(expected?: ReturnType<E>): R`)